agent_memory_server/mcp.py

```diff
@@ -450,19 +451,28 @@ async def search_long_term_memory(
             offset=offset,
         )
         results = await core_search_long_term_memory(payload)
-        results = MemoryRecordResults(
-            total=results.total,
-            memories=results.memories,
-            next_offset=results.next_offset,
+        import json as _json
+
+        return TextContent(
```

Comment:
Why is this returning TextContent? It should return MemoryRecordResults and the framework should serialize them for us after validating.